src/behavior-considered-undefined.md

```diff
@@ -23,7 +23,8 @@ code.
 </div>
 
 * Data races.
-* Dereferencing (using the `*` operator on) a dangling or unaligned raw pointer.
+* Evaluating a dereference [place expression] (`*expr`) on a raw pointer that is
```

Comment:
This feels weird to me, because a dereference is always a place expression.

Reply:
Well, it can be used as a value expression as in
```rust
let x = *expr;
```
Arguably, here it is correct to say that `*expr` is (used as) a value expression.

Reply:
We call that a "value expression context" (and "place expression context" for where a place is wanted). Is it only UB to evaluate an unaligned/dangling pointer in place expression context, or always?

Reply:
"value expression contexts" are a superset of "place expression contexts". IOW, the code above is sugar for
```rust
let x = place2value(*expr);
```
Evaluating a place expression in value expression context consists of first evaluating the place expression as normal, as then performing place-to-value conversion.

So, it is impossible to write `*expr` anywhere without it being *also* a place expression, but sometimes, it is both a place expression and a value expression (or really, `place2value(*expr)` is the value expression, but since we have no syntax for this, people tend to say that `*expr` is a value expression).

So, the answer is that it is always UB to evaluate an unaligned/dangling `*expr` since doing so always evaluates the place expression -- and then sometimes goes on performing place-to-value conversion. The important point that I hope to clarify in the docs is that it is the place expression evaluation, and *not* the place-to-value conversion, that is causing the UB.

That's the way I am thinking about this, anyway.

Reply:
> "value expression contexts" are a superset of "place expression contexts"

We have a difference of perspectives here; these two categories are distinct and split the set of expression contexts. A place expression in value expression context is allowed and has the `place2value` effect, but it's still always a place expression.

So, under my perspective, the `[place expression]` should be `, even in [place expression contexts],`.

---

> Evaluating a place expression in value expression context consists of first evaluating the place expression as normal, as then performing place-to-value conversion.

I like this sentence. I think I'll try and get it into the PR I submitted last night (#1003 )

Reply:
So, more like this?

@digama0 I'd be interested in your opinion here, is this current wording sufficiently clear? (I know you'd like the rules to be more relaxed; so do I. That is a longer process. For now the goal is to make sure that the rules as they currently are are described unambiguously.)

Reply:
LGTM